Sources/DarkModeCore/UIImage+DarkModeKit.m

```diff
@@ -36,6 +37,17 @@ + (void)dm_swizzleIsEqual {
 }
 
 + (UIImage *)dm_imageWithLightImage:(UIImage *)lightImage darkImage:(UIImage *)darkImage {
+  if (@available(iOS 13, *)) {
+    UITraitCollection *lightTraitCollection = [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleLight];
+    UITraitCollection *darkTraitCollection = [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleDark];
+
+    UIImageAsset *imageAsset = [[UIImageAsset alloc] init];
+    [imageAsset registerImage:lightImage withTraitCollection:lightTraitCollection];
```

Comment:
I've experienced issues with scaled images when using `registerImage`. In its [documentation](https://developer.apple.com/documentation/uikit/uiimageasset/1624974-register), it says
> Important
The trait collection must always contain an explicit value in its displayScale property. You may experience unexpected results from image(with:) if the trait collection does not explicitly define the desired scale.

I think we can use the `scale` property of `UIImage` here, but I'm not sure whether we should also use a scale value to retrieve images from image assets.

Reply:
is it a must to register images of different scales if it is from image assets (and can we get these images from the image property?) ?

Reply:
I don't think so. What is says is we must specify a `displayScale` value to `UITraitCollection`. Try testing with scaled images. I tried registering a 2x image and it was treated as an 1x image when retrieved.

Reply:
I think it should be sufficient to just use the image's scale property here. I tested the case scenario where there are images of different scales is in the image asset in Xcode, only the one with closest (to screen scale) scale is picked no matter what displayScale I pass to `[UIImageAsset imageWithTraitCollection:]`. I guess only best scale image is loaded when app runs.

For image asset defined in code, the user has to fetch a specific image with a configuration or trait collection, so it should already be the desired image, so I don't think we need to query for other images in the image asset.